src/services/services.ts

```diff
@@ -2072,7 +2079,8 @@ namespace ts {
             compilationSettings: CompilerOptions,
             scriptSnapshot: IScriptSnapshot,
             version: string,
-            acquiring: boolean): SourceFile {
+            acquiring: boolean,
+            scriptKind?: ScriptKind): SourceFile {
 
             const bucket = getBucketForCompilationSettings(compilationSettings, /*createIfMissing*/ true);

```

Comment:
I'm having a hard time seeing where this would occur, but I can see it's possible, and imagine this could be really hard to figure out the bug if it did :wink: 

How would we easiest handle this? We could mangle the filename, but that seems ugly. Key the file off of two properties (filename and kind)?